### PR TITLE
Upgrade support

### PR DESCRIPTION
**Pull request by ashmckenzie**
This PR does the following:

* Install resource now supports `upgrade` boolean
* Use `allow_any_instance_of(Chef::Resource)` for stubbing over `stub_command`
* Removed a few methods that were not in use
* Increases unit test coverage to 100% :)

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -7,7 +7,7 @@ group :test do
   gem 'berkshelf'
   gem 'test-kitchen'
   gem 'kitchen-vagrant'
-  gem 'chefspec'
+  gem 'chefspec', '~> 4'
```
Comment (jacknagz) [inline Gemfile]: latest chefspec uses Chef 13 right?

File: libraries/installer_helpers.rb [hunk 2]
```diff
@@ -15,8 +15,7 @@ def osquery_current_version
   end
 
   def rsyslog_legacy
-    version = Mixlib::ShellOut.new('`which rsyslogd` -v ')
-    version.run_command
+    version = shell_out!('`which rsyslogd` -v')
```
Comment (jacknagz) [inline libraries/installer_helpers.rb]: 👍

**Comment by jacknagz** [comment]
awesome! looks good to me

**Comment by jacknagz** [comment]
@ashmckenzie awesome! would you mind bumping the cookbook patch version too?

File: Gemfile (hunk 1 above)
Comment (ashmckenzie) [inline Gemfile]: Latest ChefSpec is 7.x, I added `'~> 4'` to ensure the latest 4.x is used.  I didn't go down the Chef 13 support path, 😬

**Comment by ashmckenzie** [comment]
`1.7.3` or `1.8.0` ?  IMHO, these changes introduce new functionality and warrant a minor bump :)

**Comment by jacknagz** [comment]
@ashmckenzie minor works for me

**Comment by jacknagz** [comment]
Tagged and pushed to supermarket

**Comment by ashmckenzie** [comment]
Many thanks! :)

**Pull request merged**